File: backend/src/routes/a2a.js
/**
 * A2A (Agent-to-Agent) 协议路由
 * 业务逻辑委托给 A2AService，routes 只做参数校验和响应组装
 */

const express = require('express');
const router = express.Router();
const { A2AService, A2A_MESSAGE_TYPES, A2A_TASK_STATUS } = require('../services/a2aService');
const { MultiAgentCoordinator } = require('../services/MultiAgentCoordinator');

const a2aService = new A2AService();
const coordinator = new MultiAgentCoordinator(a2aService);

const ok = (res, d) => res.json({ success: true, ...d });
const created = (res, d) => res.status(201).json({ success: true, ...d });
const fail = (res, s, m) => res.status(s).json({ success: false, error: { message: m } });

// 服务状态
router.get('/status', (req, res) => ok(res, a2aService.getStats()));

// Agent 管理
router.get('/agents', (req, res) => ok(res, { agents: a2aService.listAgents(), count: a2aService.listAgents().length }));
router.get('/agents/:agentId', (req, res) => { const a = a2aService.getAgent(req.params.agentId); if (!a) return fail(res, 404, 'Agent not found'); ok(res, { agent: a }); });
router.post('/agents/register', (req, res) => { const { id, name, type, endpoint, capabilities, metadata } = req.body; if (!id) return fail(res, 400, 'Agent ID is required'); created(res, { agent: a2aService.registerAgent({ id, name: name || id, type: type || 'general', endpoint, capabilities: capabilities || [], metadata: metadata || {} }) }); });
router.post('/agents/:agentId/unregister', (req, res) => { a2aService.unregisterAgent(req.params.agentId); ok(res, { message: 'Agent unregistered' }); });
router.post('/agents/:agentId/heartbeat', (req, res) => { a2aService.agentHeartbeat(req.params.agentId); ok(res, { timestamp: Date.now() }); });

// 消息传递
router.post('/send', (req, res) => {
  const { from, to, type, payload, taskId, priority, timeout } = req.body;
  if (!from || !to) return fail(res, 400, 'from and to are required');
  if (type === A2A_MESSAGE_TYPES.TASK_DELEGATE || type === 'task.delegate') return ok(res, a2aService.delegateTask({ from, to, title: payload?.title || 'Untitled Task', description: payload?.description || '', input: payload?.input || payload || {}, priority: priority || 0, tags: payload?.tags || [], metadata: payload?.metadata || {}, timeout: timeout || 5 * 60 * 1000 }));
  const A2AMessage = require('../services/a2aService').A2AMessage || class { constructor(o) { Object.assign(this, o); this.id = o.id || `msg_${Date.now()}`; } toJSON() { return this; } };
  const message = new A2AMessage({ type: type || 'message.send', from, to, taskId, payload: payload || {} });
  const r = a2aService.broker.send(message);
  ok(res, { messageId: message.id, message: message.toJSON() });
});

router.get('/receive', (req, res) => {
  const { agentId, limit, clear } = req.query;
  if (!agentId) return fail(res, 400, 'agentId is required');
  const msgs = a2aService.receiveMessages(agentId, { limit: parseInt(limit) || 50, includeExpired: false, clearReceived: clear === 'true' });
  a2aService.agentHeartbeat(agentId);
  ok(res, { messages: msgs.map(m => m.toJSON()), count: msgs.length, unreadCount: a2aService.getUnreadCount(agentId) });
});

router.get('/poll', (req, res) => { const { agentId, timeout } = req.query; if (!agentId) return fail(res, 400, 'agentId is required'); a2aService.pollMessages(agentId, parseInt(timeout) || 30000, req, res); });

// 任务与结果
router.post('/result/:taskId', (req, res) => { const { result, status, metadata } = req.body; if (!result) return fail(res, 400, 'result is required'); ok(res, a2aService.returnResult(req.params.taskId, result, status || A2A_TASK_STATUS.COMPLETED, metadata || {})); });
router.post('/progress/:taskId', (req, res) => { const { progress, metadata } = req.body; if (progress === undefined) return fail(res, 400, 'progress is required (0-100)'); ok(res, a2aService.sendProgress(req.params.taskId, progress, metadata || {})); });
router.post('/status/sync', (req, res) => { const { agentId, status, metadata } = req.body; if (!agentId) return fail(res, 400, 'agentId is required'); ok(res, a2aService.syncStatus(agentId, status || 'available', metadata || {})); });
router.get('/tasks/:taskId', (req, res) => { const t = a2aService.getTaskStatus(req.params.taskId); if (!t) return fail(res, 404, 'Task not found'); ok(res, { task: t }); });
router.get('/tasks', (req, res) => { const { status, from, to, limit } = req.query; const tasks = a2aService.listTasks({ status, from, to, limit: parseInt(limit) || 100 }); ok(res, { tasks, count: tasks.length }); });
router.delete('/tasks/:taskId', (req, res) => ok(res, a2aService.cancelTask(req.params.taskId)));
router.get('/unread/:agentId', (req, res) => ok(res, { unreadCount: a2aService.getUnreadCount(req.params.agentId) }));

// SSE 实时订阅
router.get('/subscribe/:agentId', (req, res) => a2aService.subscribeAgent(req.params.agentId, req, res));
router.post('/ack', (req, res) => { const { agentId, messageIds } = req.body; if (!agentId || !messageIds || !Array.isArray(messageIds)) return fail(res, 400, 'agentId and messageIds array are required'); ok(res, a2aService.ackMessages(agentId, messageIds)); });

// 协作任务
router.post('/collaborate', async (req, res) => {
  const { title, tasks, subTasks, options = {} } = req.body;
  if (!title) return fail(res, 400, { type: 'validation_error', message: 'title is required' });
  const list = tasks || subTasks;
  if (!list || !Array.isArray(list) || list.length === 0) return fail(res, 400, { type: 'validation_error', message: 'tasks must be a non-empty array' });
  for (let i = 0; i < list.length; i++) if (!list[i].task && !list[i].prompt && !list[i].description) return fail(res, 400, { type: 'validation_error', message: `Task[${i}] is missing required field: task/prompt/description` });
  ok(res, { collaboration: await coordinator.executeCollaboration(title, list, options) });
});
router.get('/collaboration/stats', (req, res) => ok(res, coordinator.getStats()));
router.get('/collaboration/:taskId', (req, res) => { const s = coordinator.getCollaborationStatus(req.params.taskId); if (!s) return fail(res, 404, 'Collaboration task not found'); ok(res, { collaboration: s }); });
router.get('/collaboration/:taskId/result', (req, res) => { const r = coordinator.getCollaborationResult(req.params.taskId); if (!r) return fail(res, 404, 'Collaboration task not found'); ok(res, r); });
router.delete('/collaboration/:taskId', (req, res) => { const c = coordinator.cancelCollaboration(req.params.taskId); if (!c) return fail(res, 404, 'Collaboration task not found or already completed'); ok(res, { message: 'Collaboration cancelled', taskId: req.params.taskId }); });

// 协作任务 SSE 订阅
router.get('/collaboration/:taskId/subscribe', (req, res) => a2aService.subscribeCollaboration(req.params.taskId, req, res));

// 任务定义
router.post('/tasks/define', (req, res) => { const t = req.body; if (!t.task && !t.prompt && !t.description) return fail(res, 400, { type: 'validation_error', message: 'task/prompt/description is required' }); const def = coordinator.createTaskDefinition(t); created(res, { task: def.toJSON() }); });
router.post('/tasks/define/batch', (req, res) => { const { tasks } = req.body; if (!tasks || !Array.isArray(tasks)) return fail(res, 400, { type: 'validation_error', message: 'tasks array is required' }); const defs = coordinator.createTaskDefinitions(tasks); created(res, { tasks: defs.map(d => d.toJSON()), count: defs.length }); });
router.get('/tasks/define/:taskId', (req, res) => { const d = coordinator.getTaskDefinition(req.params.taskId); if (!d) return fail(res, 404, 'Task definition not found'); ok(res, { task: d.toJSON() }); });
router.get('/coordination/modes', (req, res) => ok(res, { modes: { TEAM_LEADER: { value: 'team_leader', description: 'One agent orchestrates others', useCase: 'Complex hierarchical tasks' }, COLLABORATIVE: { value: 'collaborative', description: 'Agents share responsibilities', useCase: 'Parallel specialized work' }, AUTONOMOUS: { value: 'autonomous', description: 'Agents work independently', useCase: 'Independent parallel tasks' } } }));

module.exports = router;

File: backend_go/internal/handlers/a2a.go
/**
 * A2A 协议处理器
 * 处理 Agent-to-Agent 相关的 HTTP 请求
 */

package handlers

import (
	"encoding/json"
	"strconv"
	"time"

	"github.com/gin-gonic/gin"

	"github.com/ai-chat/backend_go/internal/domain/a2a"
)

// A2AHandler A2A协议处理器
type A2AHandler struct {
	service *a2a.A2AService
}

// NewA2AHandler 创建A2A处理器
func NewA2AHandler(service *a2a.A2AService) *A2AHandler {
	return &A2AHandler{service: service}
}

// RegisterRoutes 注册A2A路由
func (h *A2AHandler) RegisterRoutes(router *gin.RouterGroup) {
	// 状态和Agent管理
	router.GET("/status", h.GetStatus)
	router.GET("/agents", h.ListAgents)
	router.GET("/agents/:agentId", h.GetAgent)
	router.POST("/agents/register", h.RegisterAgent)
	router.POST("/agents/:agentId/unregister", h.UnregisterAgent)
	router.POST("/agents/:agentId/heartbeat", h.Heartbeat)

	// 消息传递
	router.POST("/send", h.SendMessage)
	router.GET("/receive", h.ReceiveMessages)
	router.GET("/poll", h.PollMessages)
	router.GET("/unread/:agentId", h.GetUnreadCount)

	// 任务管理
	router.POST("/result/:taskId", h.ReturnResult)
	router.POST("/progress/:taskId", h.SendProgress)
	router.GET("/tasks/:taskId", h.GetTaskStatus)
	router.GET("/tasks", h.ListTasks)
	router.DELETE("/tasks/:taskId", h.CancelTask)

	// SSE订阅
	router.GET("/subscribe/:agentId", h.Subscribe)

	// 协作任务
	router.POST("/collaborate", h.Collaborate)
	router.GET("/collaboration/:taskId", h.GetCollaborationStatus)
	router.DELETE("/collaboration/:taskId", h.CancelCollaboration)
	router.GET("/collaboration/stats", h.GetCollaborationStats)
}

// GetStatus 获取服务状态
// GET /api/a2a/status
func (h *A2AHandler) GetStatus(c *gin.Context) {
	stats := h.service.GetStats()
	SuccessResponse(c, stats)
}

// ListAgents 获取在线Agent列表
// GET /api/a2a/agents
func (h *A2AHandler) ListAgents(c *gin.Context) {
	agents := h.service.ListAgents()

	agentList := make([]map[string]interface{}, len(agents))
	for i, agent := range agents {
		agentList[i] = map[string]interface{}{
			"id":           agent.ID,
			"name":         agent.Name,
			"type":         agent.Type,
			"status":       agent.Status,
			"endpoint":     agent.Endpoint,
			"capabilities": agent.Capabilities,
			"metadata":     agent.Metadata,
			"lastSeen":     agent.LastSeen,
		}
	}

	SuccessResponse(c, gin.H{
		"agents": agentList,
		"count":  len(agentList),
	})
}

// GetAgent 获取单个Agent信息
// GET /api/a2a/agents/:agentId
func (h *A2AHandler) GetAgent(c *gin.Context) {
	agentID := c.Param("agentId")
	agent := h.service.GetAgent(agentID)

	if agent == nil {
		NotFoundResponse(c, "Agent不存在", "未找到对应的Agent")
		return
	}

	SuccessResponse(c, gin.H{
		"agent": map[string]interface{}{
			"id":           agent.ID,
			"name":         agent.Name,
			"type":         agent.Type,
			"status":       agent.Status,
			"endpoint":     agent.Endpoint,
			"capabilities": agent.Capabilities,
			"metadata":     agent.Metadata,
			"lastSeen":     agent.LastSeen,
		},
	})
}

// RegisterAgentRequest 注册Agent请求
type RegisterAgentRequest struct {
	ID           string                 `json:"id"`
	Name         string                 `json:"name"`
	Type         string                 `json:"type"`
	Endpoint     string                 `json:"endpoint"`
	Capabilities []string               `json:"capabilities"`
	Metadata     map[string]interface{} `json:"metadata"`
}

// RegisterAgent 注册Agent
// POST /api/a2a/agents/register
func (h *A2AHandler) RegisterAgent(c *gin.Context) {
	var req RegisterAgentRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		BadRequestResponse(c, "参数解析失败", err.Error())
		return
	}

	if req.ID == "" {
		BadRequestResponse(c, "参数错误", "Agent ID不能为空")
		return
	}

	info := a2a.AgentInfo{
		ID:           req.ID,
		Name:         req.Name,
		Type:         req.Type,
		Endpoint:     req.Endpoint,
		Capabilities: req.Capabilities,
		Metadata:     req.Metadata,
	}

	agent := h.service.RegisterAgent(info)

	SuccessResponse(c, gin.H{
		"agent": map[string]interface{}{
			"id":           agent.ID,
			"name":         agent.Name,
			"type":         agent.Type,
			"status":       agent.Status,
			"capabilities": agent.Capabilities,
		},
	})
}

// UnregisterAgent 注销Agent
// POST /api/a2a/agents/:agentId/unregister
func (h *A2AHandler) UnregisterAgent(c *gin.Context) {
	agentID := c.Param("agentId")
	h.service.UnregisterAgent(agentID)

	SuccessResponseWithMessage(c, "Agent已注销", nil)
}

// Heartbeat Agent心跳
// POST /api/a2a/agents/:agentId/heartbeat
func (h *A2AHandler) Heartbeat(c *gin.Context) {
	agentID := c.Param("agentId")
	h.service.Heartbeat(agentID)

	SuccessResponse(c, gin.H{
		"timestamp": strconv.FormatInt(SystemTimeNowMillis(), 10),
	})
}

// SendMessageRequest 发送消息请求
type SendMessageRequest struct {
	From    string                 `json:"from"`
	To      string                 `json:"to"`
	Type    string                 `json:"type"`
	Payload map[string]interface{} `json:"payload"`
	TaskID  string                 `json:"taskId"`
	Timeout int64                  `json:"timeout"`
}

// SendMessage 发送消息
// POST /api/a2a/send
func (h *A2AHandler) SendMessage(c *gin.Context) {
	var req SendMessageRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		BadRequestResponse(c, "参数解析失败", err.Error())
		return
	}

	if req.From == "" || req.To == "" {
		BadRequestResponse(c, "参数错误", "from和to不能为空")
		return
	}

	// 判断是否为任务委托
	if req.Type == "task.delegate" || req.Type == string(a2a.MessageTypeTaskDelegate) {
		task := a2a.NewA2ATask(getStringValue(req.Payload, "title"), req.From, req.To)
		if title, ok := req.Payload["title"].(string); ok {
			task.Title = title
		}
		if desc, ok := req.Payload["description"].(string); ok {
			task.Description = desc
		}
		if input, ok := req.Payload["input"].(map[string]interface{}); ok {
			task.Input = input
		}

		task.Metadata["timeout"] = req.Timeout

		result := h.service.DelegateTask(task)

		SuccessResponse(c, gin.H{
			"success":   result.Success,
			"taskId":    task.ID,
			"messageId": result.Message.ID,
			"task":      task.ToMap(),
		})
		return
	}

	// 普通消息
	msg := a2a.NewA2AMessage(a2a.MessageType(req.Type), req.From, req.To)
	msg.TaskID = req.TaskID
	msg.Payload = req.Payload

	result := h.service.SendMessage(msg)

	SuccessResponse(c, gin.H{
		"success":   result.Success,
		"messageId": result.MessageID,
	})
}

// ReceiveMessages 接收消息
// GET /api/a2a/receive
func (h *A2AHandler) ReceiveMessages(c *gin.Context) {
	agentID := c.Query("agentId")
	if agentID == "" {
		BadRequestResponse(c, "参数错误", "agentId不能为空")
		return
	}

	limit := 50
	if l := c.Query("limit"); l != "" {
		if parsed, err := strconv.Atoi(l); err == nil {
			limit = parsed
		}
	}

	clear := c.Query("clear") == "true"

	options := &a2a.ReceiveOptions{
		Limit:          limit,
		IncludeExpired: false,
		ClearReceived:  clear,
	}

	messages := h.service.ReceiveMessages(agentID, options)

	// 更新心跳
	h.service.Heartbeat(agentID)

	msgList := make([]map[string]interface{}, len(messages))
	for i, msg := range messages {
		msgList[i] = msg.ToMap()
	}

	SuccessResponse(c, gin.H{
		"messages":    msgList,
		"count":       len(msgList),
		"unreadCount": h.service.GetUnreadCount(agentID),
	})
}

// PollMessages 轮询消息
// GET /api/a2a/poll
func (h *A2AHandler) PollMessages(c *gin.Context) {
	agentID := c.Query("agentId")
	if agentID == "" {
		BadRequestResponse(c, "参数错误", "agentId不能为空")
		return
	}

	// 更新心跳
	h.service.Heartbeat(agentID)

	// 立即返回可用消息
	messages := h.service.ReceiveMessages(agentID, &a2a.ReceiveOptions{
		Limit:         50,
		ClearReceived: true,
	})

	msgList := make([]map[string]interface{}, len(messages))
	for i, msg := range messages {
		msgList[i] = msg.ToMap()
	}

	SuccessResponse(c, gin.H{
		"messages": msgList,
		"count":    len(msgList),
	})
}

// GetUnreadCount 获取未读消息数
// GET /api/a2a/unread/:agentId
func (h *A2AHandler) GetUnreadCount(c *gin.Context) {
	agentID := c.Param("agentId")
	count := h.service.GetUnreadCount(agentID)

	SuccessResponse(c, gin.H{
		"unreadCount": count,
	})
}

// ReturnResultRequest 返回结果请求
type ReturnResultRequest struct {
	Result   interface{}            `json:"result"`
	Status   string                 `json:"status"`
	Metadata map[string]interface{} `json:"metadata"`
}

// ReturnResult 返回结果
// POST /api/a2a/result/:taskId
func (h *A2AHandler) ReturnResult(c *gin.Context) {
	taskID := c.Param("taskId")

	var req ReturnResultRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		BadRequestResponse(c, "参数解析失败", err.Error())
		return
	}

	status := a2a.TaskStatus(req.Status)
	if status == "" {
		status = a2a.TaskStatusCompleted
	}

	metadata := req.Metadata
	if metadata == nil {
		metadata = make(map[string]interface{})
	}

	result := h.service.ReturnResult(taskID, req.Result, status, metadata)

	if !result.Success {
		NotFoundResponse(c, "任务不存在", result.Error)
		return
	}

	SuccessResponse(c, result)
}

// SendProgress 发送进度更新
// POST /api/a2a/progress/:taskId
func (h *A2AHandler) SendProgress(c *gin.Context) {
	taskID := c.Param("taskId")

	var req struct {
		Progress int                    `json:"progress"`
		Metadata map[string]interface{} `json:"metadata"`
	}

	if err := c.ShouldBindJSON(&req); err != nil {
		BadRequestResponse(c, "参数解析失败", err.Error())
		return
	}

	if req.Progress < 0 || req.Progress > 100 {
		BadRequestResponse(c, "参数错误", "progress必须在0-100之间")
		return
	}

	metadata := req.Metadata
	if metadata == nil {
		metadata = make(map[string]interface{})
	}

	result := h.service.SendProgress(taskID, req.Progress, metadata)

	if !result.Success {
		NotFoundResponse(c, "任务不存在", result.Error)
		return
	}

	SuccessResponse(c, gin.H{
		"messageId": result.MessageID,
	})
}

// GetTaskStatus 获取任务状态
// GET /api/a2a/tasks/:taskId
func (h *A2AHandler) GetTaskStatus(c *gin.Context) {
	taskID := c.Param("taskId")
	task := h.service.GetTaskStatus(taskID)

	if task == nil {
		NotFoundResponse(c, "任务不存在", "未找到对应的任务")
		return
	}

	SuccessResponse(c, gin.H{
		"task": task.ToMap(),
	})
}

// ListTasks 列出任务
// GET /api/a2a/tasks
func (h *A2AHandler) ListTasks(c *gin.Context) {
	status := c.Query("status")
	from := c.Query("from")
	to := c.Query("to")
	limit := 100

	if l := c.Query("limit"); l != "" {
		if parsed, err := strconv.Atoi(l); err == nil {
			limit = parsed
		}
	}

	filter := &a2a.TaskFilter{
		Status: status,
		From:   from,
		To:     to,
		Limit:  limit,
	}

	tasks := h.service.ListTasks(filter)

	taskList := make([]map[string]interface{}, len(tasks))
	for i, task := range tasks {
		taskList[i] = task.ToMap()
	}

	SuccessResponse(c, gin.H{
		"tasks": taskList,
		"count": len(taskList),
	})
}

// CancelTask 取消任务
// DELETE /api/a2a/tasks/:taskId
func (h *A2AHandler) CancelTask(c *gin.Context) {
	taskID := c.Param("taskId")
	success := h.service.CancelTask(taskID)

	if !success {
		NotFoundResponse(c, "任务不存在", "未找到对应的任务")
		return
	}

	SuccessResponseWithMessage(c, "任务已取消", nil)
}

// Subscribe SSE订阅消息
// GET /api/a2a/subscribe/:agentId
func (h *A2AHandler) Subscribe(c *gin.Context) {
	agentID := c.Param("agentId")

	// 更新心跳
	h.service.Heartbeat(agentID)

	// 设置SSE头
	c.Header("Content-Type", "text/event-stream")
	c.Header("Cache-Control", "no-cache")
	c.Header("Connection", "keep-alive")

	// 订阅消息
	ch := h.service.Subscribe(agentID)
	defer h.service.Unsubscribe(agentID)

	// 定期心跳
	ticker := time.NewTicker(30 * time.Second)
	defer ticker.Stop()

	// 定期检查新消息
	pollTicker := time.NewTicker(2 * time.Second)
	defer pollTicker.Stop()

	// 初始连接确认
	c.SSEvent("connected", gin.H{"agentId": agentID})
	c.Writer.Flush()

	for {
		select {
		case <-c.Request.Context().Done():
			return
		case <-ticker.C:
			h.service.Heartbeat(agentID)
			c.SSEvent("heartbeat", gin.H{"timestamp": strconv.FormatInt(SystemTimeNowMillis(), 10)})
			c.Writer.Flush()
		case msg := <-ch:
			c.SSEvent("message", msg.ToMap())
			c.Writer.Flush()
		case <-pollTicker.C:
			messages := h.service.ReceiveMessages(agentID, &a2a.ReceiveOptions{
				Limit:         10,
				ClearReceived: true,
			})
			for _, msg := range messages {
				c.SSEvent("message", msg.ToMap())
				c.Writer.Flush()
			}
		}
	}
}

// Collaborate 协作任务处理
// POST /api/a2a/collaborate
func (h *A2AHandler) Collaborate(c *gin.Context) {
	var req struct {
		Title    string                   `json:"title"`
		SubTasks []map[string]interface{} `json:"subTasks"`
		Options  map[string]interface{}   `json:"options"`
	}

	if err := c.ShouldBindJSON(&req); err != nil {
		BadRequestResponse(c, "参数解析失败", err.Error())
		return
	}

	if req.Title == "" {
		BadRequestResponse(c, "参数错误", "title不能为空")
		return
	}

	// 简化的协作任务实现
	SuccessResponse(c, gin.H{
		"collaboration": map[string]interface{}{
			"title":    req.Title,
			"subTasks": req.SubTasks,
			"status":   "processing",
		},
	})
}

// GetCollaborationStatus 获取协作状态
// GET /api/a2a/collaboration/:taskId
func (h *A2AHandler) GetCollaborationStatus(c *gin.Context) {
	taskID := c.Param("taskId")

	SuccessResponse(c, gin.H{
		"collaboration": map[string]interface{}{
			"taskId": taskID,
			"status": "completed",
		},
	})
}

// CancelCollaboration 取消协作
// DELETE /api/a2a/collaboration/:taskId
func (h *A2AHandler) CancelCollaboration(c *gin.Context) {
	taskID := c.Param("taskId")

	SuccessResponseWithMessage(c, "协作已取消", gin.H{"taskId": taskID})
}

// GetCollaborationStats 获取协作统计
// GET /api/a2a/collaboration/stats
func (h *A2AHandler) GetCollaborationStats(c *gin.Context) {
	SuccessResponse(c, gin.H{
		"activeCollaborations":    0,
		"completedCollaborations": 0,
	})
}

// SystemTimeNowMillis 获取当前毫秒时间戳
func SystemTimeNowMillis() int64 {
	return time.Now().UnixMilli()
}

// getStringValue 安全获取字符串值
func getStringValue(m map[string]interface{}, key string) string {
	if v, ok := m[key].(string); ok {
		return v
	}
	return ""
}

// MarshalJSON 安全JSON序列化
func MarshalJSON(v interface{}) string {
	data, _ := json.Marshal(v)
	return string(data)
}
